bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -53,37 +57,45 @@ object BagsManager {
                 unmuteObservation(observation)
               }
             }
-            taskComplete(this, success = true)
+            taskSuccess(this)
 
           // We don't want to print the stack trace if the host is simply unreachable.
           // This is reported only as a GenericError in a CatalogException, unfortunately.
           case Failure(CatalogException((e: GenericError) :: _)) =>
-            LOG.warning(s"BAGS lookup for observation=${observation.getObservationID} failed: ${e.msg}")
-            taskComplete(this, success = false)
+            LOG.warning(s"$bagsIdMsg failed: ${e.msg}")
+            taskFail(this)
+
+          // If we timed out, we don't want to delay.
+          case Failure(ex: TimeoutException) =>
+            LOG.warning(s"$bagsIdMsg failed: ${ex.getMessage}")
+            taskFail(this, timedOut = true)
 
           // For all other exceptions, print the full stack trace.
           case Failure(ex) =>
-            LOG.log(Level.WARNING, s"BAGS lookup for observation=${observation.getObservationID} failed.", ex)
-            taskComplete(this, success = false)
+            LOG.log(Level.WARNING, s"$bagsIdMsg} failed.", ex)
+            taskFail(this)
         }
       }
 
 
-      LOG.info(s"Performing BAGS lookup for observation=${observation.getObservationID}")
+      LOG.info(s"Performing $bagsIdMsg.")
       curCtx.foreach { obsCtx =>
         AgsRegistrar.currentStrategy(obsCtx).foreach { strategy =>
           if (GuideStarSupport.hasGemsComponent(observation)) {
             lookup((x: GOption[GemsGuideStars]) => x.asScalaOpt)(GemsGuideStarWorker.applyResults(_, _, true))(Try(GemsGuideStarWorker.findGuideStars(obsCtx)))
           } else {
+            // This future always completes, either with results or timeout of 30s, so we wait for it with a long timeout
+            // to let it do its thing.
             val fut = strategy.select(obsCtx, OT.getMagnitudeTable)
-            fut.onComplete(lookup(identity[Option[AgsStrategy.Selection]])(GuideStarWorker.applyResults))
+            val futDone = Await.ready(fut, LookupTimeout)
```

Comment:
This is the result of trying to mix Java `Thread`s with Scala `Future`s so that we can deal with GeMS (Java) and AGS (Scala) in the same way, and I'm really not sure how else to handle it.

Essentially, we want the `Future` to complete within the Java `Thread` that executes it before the `Thread` terminates. I realize that this is not the way that `Futures` are supposed to be used, generally, and is sort of an abuse of the concept, but I don't know how else to make the thread wait on the future to complete before requeueing itself for work.